source/beamline/CLS/CLSShutters.cpp

```diff
@@ -14,7 +19,88 @@ CLSShutters::CLSShutters(const QString &name, QObject *parent) :
 
 CLSShutters::~CLSShutters()
 {
+	shuttersBeamOnOrderMap_.clear();
+}
+
+AMListAction3* CLSShutters::createBeamOnActionList()
```

Comment:
Is there something beam on/off specific about these action lists? Or are they just opening/closing the shutters and valves? If it's the latter case, we could think about them instead in terms of a shutters move action (eg. `moveToOpen()`, `moveToClosed()`) and reimplement `AMPseudoMotorControl::createMoveAction(double)`.